mcs/resource_mcs_db_instance.go

```diff
@@ -0,0 +1,695 @@
+				Computed: false,
+				ForceNew: true,
+			},
+
+			"public_access": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: false,
+				ForceNew: true,
+			},
+
+			"keypair": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: false,
+				ForceNew: true,
+			},
+
+			"disk_autoexpand": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: false,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"autoexpand": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							ForceNew: false,
+						},
+						"max_disk_size": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							ForceNew: false,
+						},
+					},
+				},
+			},
+
+			"capabilities": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: false,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"settings": {
+							Type:     schema.TypeMap,
+							Optional: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+						},
+					},
+				},
+			},
+		},
+		CustomizeDiff: customdiff.All(
+			customdiff.ValidateChange("size", func(old, new, meta interface{}) error {
+				if new.(int) < old.(int) {
+					return fmt.Errorf("the new volume size %d must be larger than the current volume size of %d", new.(int), old.(int))
+				}
+				return nil
+			}),
+			customdiff.ValidateChange("wal_volume", func(old, new, meta interface{}) error {
+				if len(old.([]interface{})) == 0 {
+					return nil
+				}
+
+				walVolumeOptsNew, err := extractDatabaseInstanceWalVolume(new.([]interface{}))
+				if err != nil {
+					return fmt.Errorf("unable to determine mcs_db_instance wal_volume")
+				}
+
+				walVolumeOptsOld, err := extractDatabaseInstanceWalVolume(old.([]interface{}))
+				if err != nil {
+					return fmt.Errorf("unable to determine mcs_db_instance wal_volume")
+				}
+
+				if walVolumeOptsNew.Size < walVolumeOptsOld.Size {
+					return fmt.Errorf("the new wal volume size %d must be larger than the current volume size of %d", walVolumeOptsNew.Size, walVolumeOptsOld.Size)
+				}
+				return nil
+			}),
+		),
+	}
+}
+
+func resourceDatabaseInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(Config)
+	DatabaseV1Client, err := config.DatabaseV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating OpenStack database client: %s", err)
+	}
+
+	size := d.Get("size").(int)
+	createOpts := &DbInstanceCreateOpts{
+		FlavorRef:        d.Get("flavor_id").(string),
+		Name:             d.Get("name").(string),
+		Volume:           &Volume{Size: &size, VolumeType: d.Get("volume_type").(string)},
+		ReplicaOf:        d.Get("replica_of").(string),
+		AvailabilityZone: d.Get("availability_zone").(string),
+		PublicAccess:     d.Get("public_access").(bool),
```

Comment:
in public API it's called `allow_remote_access`. is it a good idea to change it in TF?

Reply:
The question is for Lazarenko and Rann. The reference tf provided by Dmitry contains public_access exactly.

Reply:
Talked with them. Agreed on `floating_ip_enabled` - same as in Magnum

Reply:
Fixed